src/index.ts

```diff
@@ -0,0 +1,173 @@
+ */
+export const requiredVar = (varName: string): string => {
+	const s = optionalVar(varName);
+	if (s == null) {
+		throw new Error(`Missing environment variable: ${varName}`);
+	}
+	return s;
+};
+
+/**
+ * Fetches the contents of an env var if it exists and is not empty, otherwise returning the specified default value,
+ * this means that if no default is specified then `''` will be normalized to `undefined`
+ */
+export function optionalVar<R>(varName: string, defaultValue: R): string | R;
+export function optionalVar<R>(
+	varName: string,
+	defaultValue?: undefined,
+): string | undefined;
+export function optionalVar<R>(
+	varName: string,
+	defaultValue?: R,
+): string | R | undefined {
+	const s = process.env[varName];
+	if (s == null || s === '') {
+		return defaultValue;
+	}
+	return s;
+}
+
+/**
+ * Checks that a string is a valid int and returns the number if so
+ */
+export const checkInt = (s: string): number | undefined => {
+	s = s.trim();
+	if (/[^0-9]/.test(s)) {
+		return;
+	}
+	const i = parseInt(s, 10);
+	if (!Number.isFinite(i)) {
+		return;
+	}
+	return i;
+};
+
+/**
+ * Fetches the contents of an env var, ensuring it is a valid integer if set,
+ * if it is not set then instead it returns the specified default value
+ */
+export function intVar(varName: string): number;
+export function intVar<R>(varName: string, defaultValue: R): number | R;
+export function intVar<R>(varName: string, defaultValue?: R): number | R {
+	if (arguments.length === 1) {
+		requiredVar(varName);
+	}
+
+	const s = optionalVar(varName);
+	if (s == null) {
+		return defaultValue!;
+	}
+
+	const i = checkInt(s);
+	if (i === undefined) {
+		throw new Error(`${varName} must be a valid number if set`);
+	}
+	return i;
+}
+
+/**
+ * Fetches the contents of an env var, ensuring it is a valid boolean (`'true'`/`'false'`) if set,
+ * if it is not set then instead it returns the specified default value
+ */
+export function boolVar(varName: string): boolean;
+export function boolVar<R>(varName: string, defaultValue: R): boolean | R;
+export function boolVar<R>(varName: string, defaultValue?: R): boolean | R {
+	if (arguments.length === 1) {
+		requiredVar(varName);
+	}
+
+	const s = optionalVar(varName);
+	if (s == null) {
+		return defaultValue!;
+	}
+	if (s === 'false') {
+		return false;
+	}
+	if (s === 'true') {
+		return true;
+	}
+	throw new Error(
+		`Invalid value for boolean var '${varName}', got '${s}', expected 'true' or 'false'`,
+	);
+}
+
+export type HostPort = { host: string; port: number };
+/**
+ * Splits an env var in the format of `${host1}:${port1}, ${host2}:${port2}, ...`
+ * into an array of individual host/port pairings
+ */
+export const hostPortsVar = (
+	varName: string,
+	defaultHosts?: HostPort[],
+): HostPort[] => {
+	const hostPairs = optionalVar(varName);
+	if (hostPairs == null) {
+		if (defaultHosts == null) {
+			throw new Error(`Missing environment variable: ${varName}`);
+		}
+		return defaultHosts;
+	}
```

Comment:
Can't this be solved in the same way as L144:147

Reply:
Actually L144:147 should have been being handled the same way as this, specifically the benefit of this is that we only apply the parsing to the env var and *not* the default value, so if the var does not exist we return it directly as the user passed